02_calculator/calculator.rb

```diff
@@ -9,7 +9,26 @@ def subtract(first, second)
 end
 
 def sum(array)
-  summ = 0
-  array.each { |x| summ += x }
-  summ
+  array.reduce(0) { |summ, n| summ + n }
+end
+
+def multiply(numbers)
+  numbers.reduce(1) { |prod, n| prod * n }
+end
+
+def power(number, power)
+  number**power
+end
+
+def factorial(number)
+  response = 1
+  if number == 0
+    return response
```

Comment:
How can we write the loop statement so that the 0 case is not an exception?

Reply:
change condition from "!= 1" to "> 1" and at the end the value will be the "response" variable